src/traits/keys.rs

```diff
@@ -27,6 +28,22 @@ pub trait PublicKeyParts {
     fn n_bits_precision(&self) -> u32 {
         self.n().bits_precision()
     }
+
+    /// Returns the big endian serialization of the modulus of the key
+    fn n_bytes(&self) -> Box<[u8]> {
+        let modulus = self.n();
+        let modulus_zeros = modulus.leading_zeros() as usize / 8;
+        let modulus = modulus.to_be_bytes();
+        (&modulus[modulus_zeros..]).into()
```

Comment:
This is the logic it would be good to extract into `BoxedUint` (possibly e.g. `to_be_bytes_trimmed`?)